### PR TITLE
#174797523 Build one-way trip request

### PR DESCRIPTION
**Pull request by Adadie**
#### What does this PR do?
This PR creates an endpoint for user one-way trip booking request.
#### Description of Task to be completed?
- create booking model and  migration
- create oneway trip booking endpoint
- create endpoint validations
#### How should this be manually tested?

- Clone the repo git clone https://github.com/atlp-rwanda/script-heroes-bn-backend.git
- Switch your working directory `cd script-heroes-bn-backend`
- Checkout the branch `git checkout ft-one-way-trip-#174797523`
- Setup all variables in .env file
- Install all dependencies npm install
- Create migrations `sequelize db:migrate`
- Start server npm run dev
- Send a POST request to localhost:3000/api/trips/oneway

#### What are the endpoints paths?

- POST: `/api/trips/oneway`
- GET ONE: `/api/trips/oneway/:tripId`
- GET ALL: `/api/trips/oneway`
- UPDATE: `/api/trips/oneway/:tripId`
- DELETE: `/api/trips/oneway/:tripId`

#### What is the model schema example?
```{
        "origin": 24,
        "destination": 22,
        "from": "2020-10-02",
        "till": "2020-10-04",
        "accomodationId": 1,
        "travelReasons": "You Know Your Reasons"
}
```

#### Any more necessary information?

- Make sure the requester logged in has a valid line manager that matches the linemager id in the user's table.
- Make sure location and accomodation ids exist in the database.
#### What are the relevant pivotal tracker stories?
[#174797523](https://www.pivotaltracker.com/story/show/174797523)

#### Screenshots

Screen Shot 2020-10-22 at 14 48 58
Screen Shot 2020-10-22 at 14 49 12
Screen Shot 2020-10-22 at 14 50 10
Screen Shot 2020-10-22 at 14 51 30
Screen Shot 2020-10-22 at 14 51 59

### CODE REVIEW

File: src/database/models/trip.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    {
+      requestId: {
+        type: DataTypes.STRING,
+        onDelete: 'CASCADE',
+        onUpdate: 'CASCADE'
+      },
+      accomodationId: DataTypes.STRING,
+      departFrom: DataTypes.STRING,
+      departTo: DataTypes.STRING,
+      departDate: DataTypes.STRING,
```
Comment (kipropbrian) [inline src/database/models/trip.js]: Check with @elvisiraguha . departTo is a location, thus should be an int

Comment (Adadie) [reply src/database/models/trip.js]: Okay, thanks for the heads up @kipropbrian

**Comment by valens10** [comment]
You need to rebase

**Review by dushimeemma — APPROVED** [review]
Smell's good to me but try to work on tests to increase coverage

**Review by elvisiraguha — APPROVED** [review]

**Review by valens10 — APPROVED** [review]

**Review by b0nbon1 — APPROVED** [review]
Looks good

**Pull request merged**